tests/test_agent.py

```diff
@@ -3092,8 +3092,8 @@ def test_image_url_serializable_missing_media_type():
                                 'url': 'https://example.com/chart.jpeg',
                                 'force_download': False,
                                 'vendor_metadata': None,
-                                '_media_type': None,
                                 'kind': 'image-url',
+                                'media_type': 'image/jpeg',
```

Comment:
The one thing with this is that if you don't have `media_type` defined, FileUrl will infer from the URL, but then once you serialize / deserialize it, it will only read from `media_type` property, not inferring from the URL. I don't think that's a big deal because the media type shouldn't change, but I wonder if just having media_type default based on the provided URL vs using the property would be better so it's more consistent? I've done this before with pydantic where I'll have a field depend on another field but I can't recall the exact syntax.